<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from table_sw_one.xacro             | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="table" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Import wall socket -->
  <link name="link_socket">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://objects/meshes/socket_one/socket_one.dae"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0666666666667" ixy="0.0" ixz="0.0" iyy="0.0133666666667" iyz="0.0" izz="0.0533666666667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://objects/meshes/socket_one/socket_one.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="joint_fixed_socket" type="fixed">
    <origin rpy="0 0 -1.5707963268" xyz="-0.01 0 0"/>
    <parent link="link_socket"/>
    <child link="link_wall"/>
  </joint>
  <link name="link_wall">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://objects/meshes/wall.dae"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0666666666667" ixy="0.0" ixz="0.0" iyy="0.0133666666667" iyz="0.0" izz="0.0533666666667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.8 0.4 0.02"/>
      </geometry>
    </collision>
  </link>
  <joint name="link_table_wall" type="fixed">
    <!-- "${0.24 - 0.03} 0 ${0.2 + 0.04}" r-->
    <origin rpy="0 0 1.5707963268" xyz="0 -0.21 -0.24"/>
    <parent link="link_wall"/>
    <child link="average_table_link"/>
  </joint>
  <!-- Import table -->
  <gazebo reference="table_link">
    <material>Gazebo/Wood</material>
  </gazebo>
  <gazebo reference="front_right_leg">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="front_left_leg">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="rear_left_leg">
    <material>Gazebo/Grey</material>
  </gazebo>
  <gazebo reference="rear_right_leg">
    <material>Gazebo/Grey</material>
  </gazebo>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.6 0.6 0.6 1.0"/>
  </material>
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 1.0 1"/>
  </material>
  <material name="Wood">
    <color rgba="1 0.88 0.56 1"/>
  </material>
  <link name="average_table_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <geometry>
        <box size="0.5 0.9 0.04"/>
      </geometry>
      <material name="Wood">
        <color rgba="1 0.88 0.56 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0883333333333" ixy="0.0" ixz="0.0" iyy="0.0676333333333" iyz="0.0" izz="0.0209666666667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.5 0.9 0.04"/>
      </geometry>
    </collision>
  </link>
  <link name="front_left_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
      <material name="Grey">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0111104166667" ixy="0.0" ixz="0.0" iyy="1.66666666667e-05" iyz="0.0" izz="0.0111104166667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
    </collision>
  </link>
  <joint name="link_front_left_leg" type="fixed">
    <origin rpy="0 0 0" xyz="0.24 0.44 -0.385"/>
    <parent link="average_table_link"/>
    <child link="front_left_leg"/>
  </joint>
  <link name="front_right_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
      <material name="Grey">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0111104166667" ixy="0.0" ixz="0.0" iyy="1.66666666667e-05" iyz="0.0" izz="0.0111104166667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
    </collision>
  </link>
  <joint name="link_front_right_leg" type="fixed">
    <origin rpy="0 0 0" xyz="0.24 -0.44 -0.385"/>
    <parent link="average_table_link"/>
    <child link="front_right_leg"/>
  </joint>
  <link name="rear_left_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
      <material name="Grey">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0111104166667" ixy="0.0" ixz="0.0" iyy="1.66666666667e-05" iyz="0.0" izz="0.0111104166667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
    </collision>
  </link>
  <joint name="link_rear_left_leg" type="fixed">
    <origin rpy="0 0 0" xyz="-0.24 0.44 -0.385"/>
    <parent link="average_table_link"/>
    <child link="rear_left_leg"/>
  </joint>
  <link name="rear_right_leg">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
      <material name="Grey">
        <color rgba="0.4 0.4 0.4 1"/>
      </material>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1"/>
      <inertia ixx="0.0111104166667" ixy="0.0" ixz="0.0" iyy="1.66666666667e-05" iyz="0.0" izz="0.0111104166667"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.02 0.02 0.73"/>
      </geometry>
    </collision>
  </link>
  <joint name="link_rear_right_leg" type="fixed">
    <origin rpy="0 0 0" xyz="-0.24 -0.44 -0.385"/>
    <parent link="average_table_link"/>
    <child link="rear_right_leg"/>
  </joint>
</robot>

